File: OneCodeTeam/Desktop Application developed in C# to upload a file using FTPWebRequest/[VB]-Desktop Application developed in C# to upload a file using FTPWebRequest/VB/FileUpload_FTPWebRequest_VB/Form1.vb

'+++++++++++++++DISCLAIMER+++++++++++++++++++++++++++++
'--------------------------------------------------------------------------------- 
'The sample scripts are Not supported under any Microsoft standard support program Or service. The sample scripts are provided As Is without warranty  
'Of any kind. Microsoft further disclaims all implied warranties including, without limitation, any implied warranties Of merchantability Or Of fitness For 
'a particular purpose. The entire risk arising out Of the use Or performance Of the sample scripts And documentation remains With you. In no Event shall 
'Microsoft, its authors, Or anyone Else involved In the creation, production, Or delivery Of the scripts be liable For any damages whatsoever (including, 
'without limitation, damages For loss Of business profits, business interruption, loss Of business information, Or other pecuniary loss) arising out Of the use 
'Of Or inability To use the sample scripts Or documentation, even If Microsoft has been advised Of the possibility Of such damages 
'---------------------------------------------------------------------------------



Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports System.Net
Imports System.IO

Public Class Form1


    Private Sub btnBrowse_Click(sender As Object, e As EventArgs) Handles Button1.Click

        Try
            'Browse the file to be uploaded
            Dim defaultPath As String = "C:\traces\"
            OpenFileDialog1.FileName = Me.txtFilename.Text
            OpenFileDialog1.Filter = "All Files (*.pdf)|*.pdf"
            OpenFileDialog1.InitialDirectory = defaultPath
            If OpenFileDialog1.ShowDialog() = DialogResult.OK Then
                'update the TextBox with selected File's FileName property
                Me.txtFilename.Text = OpenFileDialog1.FileName
            End If
            Me.Button1.Enabled = Me.txtFilename.Text.Length > 0
        Catch ex As Exception
            MessageBox.Show("Error selecting file" + Environment.NewLine + ex.Message)
        End Try

    End Sub

    Private Sub button1_Click(sender As Object, e As EventArgs) Handles Button2.Click
        If Not File.Exists(Me.txtFilename.Text) Then
            MessageBox.Show("Please select a valid file for upload")
        Else

            Dim sourcefilepath As [String] = Me.txtFilename.Text
            Dim fileName As [String] = Path.GetFileName(Me.txtFilename.Text)

            Dim ftpusername As [String] = "Admin"
            Dim ftppassword As [String] = "Pa$$w0rd"

            'Using the System.NET FTPWebRequest class 

            Dim ftp As FtpWebRequest = DirectCast(FtpWebRequest.Create("ftp://localhost/httpdocs/" + DocumentDirectory.Text + "/" + fileName), FtpWebRequest)

            'Enabling the Passive mode FTP with SSL

            ftp.UsePassive = True
            ftp.EnableSsl = True
            ftp.Method = WebRequestMethods.Ftp.ListDirectory

            'Validating the remote server certificate
            ServicePointManager.ServerCertificateValidationCallback = New System.Net.Security.RemoteCertificateValidationCallback(AddressOf AcceptAllCertifications)

            ftp.Credentials = New NetworkCredential(ftpusername, ftppassword)

            ftp.KeepAlive = True
            ftp.UseBinary = True
            ftp.Method = WebRequestMethods.Ftp.UploadFile

            'Read the File to be uploaded to the server
            Dim fs As FileStream = File.OpenRead(sourcefilepath)
            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
            fs.Read(buffer, 0, buffer.Length)
            fs.Close()

            Dim ftpstream As Stream = ftp.GetRequestStream()
            ftpstream.Write(buffer, 0, buffer.Length)
            ftpstream.Close()


            progressBar1.Value = 100
        End If
    End Sub

    Public Function AcceptAllCertifications(sender As Object, certification As System.Security.Cryptography.X509Certificates.X509Certificate, chain As System.Security.Cryptography.X509Certificates.X509Chain, sslPolicyErrors As System.Net.Security.SslPolicyErrors) As Boolean
        Return True
    End Function

    'List of Radio Button to select the folder where the file needs to be uploaded
    Private Sub RadioButton1_CheckedChanged(sender As Object, e As EventArgs) Handles ProcedureDocuments.CheckedChanged
        DocumentDirectory.Text = "ProcedureDocuments"
    End Sub

    Private Sub RadioButton2_CheckedChanged(sender As Object, e As EventArgs) Handles PlanningDocuments.CheckedChanged
        DocumentDirectory.Text = "PlanningDocuments"
    End Sub

    Private Sub RadioButton3_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton3.CheckedChanged
        DocumentDirectory.Text = "BulletinBoardDocuments"
    End Sub

    Private Sub RadioButton4_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton4.CheckedChanged
        DocumentDirectory.Text = "GoverningDocuments"
    End Sub

    Private Sub RadioButton5_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton5.CheckedChanged
        DocumentDirectory.Text = "FinancialDocuments"
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub RadioButton7_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton7.CheckedChanged
        DocumentDirectory.Text = "MeetingDocuments"
    End Sub

    Private Sub RadioButton9_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton9.CheckedChanged
        DocumentDirectory.Text = "InvoiceDocuments"
    End Sub

    Private Sub RadioButton8_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton8.CheckedChanged
        DocumentDirectory.Text = "ContractDocuments"
    End Sub

    Private Sub RadioButton10_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton10.CheckedChanged
        DocumentDirectory.Text = "Directories"
    End Sub
End Class

File: OneCodeTeam/Desktop Application developed in C# to upload a file using FTPWebRequest/[C#]-Desktop Application developed in C# to upload a file using FTPWebRequest/C#/FileUpload_FTPWebRequest/Form1.cs

//+++++++++++++++DISCLAIMER+++++++++++++++++++++++++++++
//--------------------------------------------------------------------------------- 
//The sample scripts are not supported under any Microsoft standard support program or service. The sample scripts are provided AS IS without warranty  
//of any kind. Microsoft further disclaims all implied warranties including, without limitation, any implied warranties of merchantability or of fitness for 
//a particular purpose. The entire risk arising out of the use or performance of the sample scripts and documentation remains with you. In no event shall 
//Microsoft, its authors, or anyone else involved in the creation, production, or delivery of the scripts be liable for any damages whatsoever (including, 
//without limitation, damages for loss of business profits, business interruption, loss of business information, or other pecuniary loss) arising out of the use 
//of or inability to use the sample scripts or documentation, even if Microsoft has been advised of the possibility of such damages 
//---------------------------------------------------------------------------------


using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Net;
using System.IO;

namespace FileUpload_FTPWebRequest
{
    public partial class Form1 : Form
    {
        
        public Form1()
        {
            
            InitializeComponent();
        }

        protected override void OnLoad(EventArgs e)
        {
            //Disable the Upload and Stop button untill the file is selected for uploading
            base.OnLoad(e);
            button1.Enabled = false;
          
        }

        private void button1_Click(object sender, EventArgs e)
        {

            if (!File.Exists(this.txtFilename.Text))
                MessageBox.Show("Please select a valid file for upload");
            else
            {
              
                String sourcefilepath = this.txtFilename.Text;   
                String fileName= Path.GetFileName(this.txtFilename.Text);
               
                String ftpusername = "admin";
                String ftppassword = "P@$$W0rd";

                //Using the System.NET FTPWebRequest class
                FtpWebRequest ftp = (FtpWebRequest)FtpWebRequest.Create("ftp://localhost/httpdocs/" + DocumentDirectory.Text + "/" + fileName);

                //Enabling the Passive mode FTP with SSL
                ftp.UsePassive = true;
                ftp.EnableSsl = true;
                ftp.Method = WebRequestMethods.Ftp.ListDirectory;

                //Validating the remote server certificate
                ServicePointManager.ServerCertificateValidationCallback = new System.Net.Security.RemoteCertificateValidationCallback(AcceptAllCertifications);

                ftp.Credentials = new NetworkCredential(ftpusername, ftppassword);

                ftp.KeepAlive = true;
                ftp.UseBinary = true;
                ftp.Method = WebRequestMethods.Ftp.UploadFile;

                //Read the File to be uploaded to the server
                FileStream fs = File.OpenRead(sourcefilepath);
                byte[] buffer = new byte[fs.Length];
                fs.Read(buffer, 0, buffer.Length);
                fs.Close();

                Stream ftpstream = ftp.GetRequestStream();
                ftpstream.Write(buffer, 0, buffer.Length);
                ftpstream.Close();
                
                progressBar1.Value =100;
            }
        }

        public bool AcceptAllCertifications(object sender, System.Security.Cryptography.X509Certificates.X509Certificate certification, System.Security.Cryptography.X509Certificates.X509Chain chain, System.Net.Security.SslPolicyErrors sslPolicyErrors)
        {
            return true;
        }

        //List of Radio Button to select the folder where the file needs to be uploaded
        public void radioButton1_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "ProcedureDocuments";
        }

        public void radioButton2_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "PlanningDocuments";
        }

        private void label2_Click(object sender, EventArgs e)
        {

        }

        

        private void btnBrowse_Click(object sender, EventArgs e)
        {
            try
            {
                //Browse the file to be uploaded
                string defaultPath= @"C:\traces\";
                openFileDialog1.FileName = this.txtFilename.Text;
                openFileDialog1.Filter = "All Files (*.pdf)|*.pdf";
                openFileDialog1.InitialDirectory = defaultPath;
                if (openFileDialog1.ShowDialog() == DialogResult.OK)
                {
                    //update the TextBox with selected File's FileName property
                    this.txtFilename.Text = openFileDialog1.FileName;
                }
                this.button1.Enabled = this.txtFilename.Text.Length > 0;
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error selecting file" + Environment.NewLine + ex.Message);
            }

        }

        private void label4_Click(object sender, EventArgs e)
        {

        }

        private void radioButton3_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "BulletinBoardDocuments";
        }

        private void radioButton4_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "GoverningDocuments";
        }

        private void radioButton5_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "FinancialDocuments";
        }

        private void radioButton6_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "MeetingDocuments";
        }

        private void radioButton7_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "LegalDocuments";
        }

        private void radioButton8_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "ContractDocuments";
        }

        private void radioButton9_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "InvoiceDocuments";
        }

        private void radioButton10_CheckedChanged(object sender, EventArgs e)
        {
            DocumentDirectory.Text = "Directories";
        }

    }
}
